lib/streams/join-stream.js

```diff
@@ -0,0 +1,33 @@
+var Transform = require('../transform');
+var inherits = require('util').inherits;
+
+/**
+ * JoinStream joins the incoming object stream into a data stream
+ * NOTE: writableObjectMode will always be true and readableObjectMode will always be false
+ *
+ * @class JoinStream
+ * @constructor
+ * @extends Transform
+ * @param {Object} [options] - Stream options object
+ */
+function JoinStream(options) {
```

Comment:
This is effectively the same thing as the interspersing stream with a blank delimiter.  Moreover, JoinStream implies an analogy to Array.join(), which takes a delimiter.  I'd suggest combining the two streams, and have the default delimiter be emptystring.